flows/src/main/java/com/softwaremill/jox/flows/Flow.java

```diff
@@ -1296,139 +1204,218 @@ public Flow<T> alsoToTap(Sink<T> other) {
      */
-    public InputStream runToInputStream(UnsupervisedScope scope) {
-        Source<byte[]> ch = this
-                .map(t -> {
-                    if (t instanceof byte[] bytes) {
-                        return bytes;
-                    } else {
-                        throw new IllegalArgumentException("requirement failed: method can be called only on flow containing byte[]");
-                    }
-                })
-                .runToChannel(scope);
-
-        return new InputStream() {
-            private ByteArrayIterator currentChunk = ByteArrayIterator.empty();
-
-            @Override
-            public int read() {
-                try {
-                    if (!currentChunk.hasNext()) {
-                        Object result = ch.receiveOrClosed();
-                        if (result instanceof ChannelDone) {
-                            return -1;
-                        } else if (result instanceof ChannelError error) {
-                            throw error.toException();
-                        } else {
-                            byte[] chunk = (byte[]) result;
-                            currentChunk = new ByteArrayIterator(chunk);
-                        }
-                    }
-                    return currentChunk.next() & 0xff; // Convert to unsigned
-                } catch (InterruptedException e) {
-                    throw new RuntimeException(e);
-                }
-            }
-
-            @Override
-            public int available() {
-                return currentChunk.available();
-            }
-        };
+    public Publisher<T> toPublisher(Scope scope) {
+        return new FromFlowPublisher<>(scope, last);
     }
 
+    // endregion
+
+    // region ByteFlow
+
+    public interface ByteChunkMapper<T> extends Function<T, ByteChunk> {}
+    public interface ByteArrayMapper<T> extends Function<T, byte[]> {}
+
     /**
-     * Writes content of this flow to an {@link java.io.OutputStream}.
+     * Converts a flow of `byte[]` or {@link ByteChunk} into a dedicated Flow type {@link ByteFlow}.
      *
-     * @param outputStream
-     *   Target `OutputStream` to write to. Will be closed after finishing the process or on error.
+     * @throws IllegalArgumentException if the flow does not contain `byte[]` or {@link ByteChunk} elements.
      */
-    public void runToOutputStream(OutputStream outputStream) throws Exception {
-        try {
-            last.run(t -> {
-                if (t instanceof byte[] chunk) {
-                    outputStream.write(chunk);
-                } else {
-                    throw new IllegalArgumentException("requirement failed: method can be called only on flow containing byte[]");
-                }
-            });
-            close(outputStream, null);
-        } catch (Exception e) {
-            close(outputStream, e);
-            throw e;
-        }
+    @SafeVarargs
+    public final ByteFlow toByteFlow(T... args) {
```

Comment:
what are the `args` for here?

Reply:
It's supposed to be a workaround to be able to get the Class of T and verify if it's byte[] or ByteChunk. 
I'll take a look if we can go without it

Reply:
Ok, I was able to move it to constructor of ByteFlow, but unfortunately we still need it to verify transition